### ISSUE TITLE
tspOrder fails when cluster has one marker

### ISSUE THREAD

**Issue by greymonroe** [mckaylab/TSPmap#3]
inexplicable error (ie havent looked into it when trying to order markers with TSPorder). I determined it is caused when there is a cluster of size 1 in the group. 

Error in UseMethod("as.TSP") : 
  no applicable method for 'as.TSP' applied to an object of class "c('double', 'numeric')"

**Comment by LovellHAGSC** [comment]
yes - you can't solve a TSP with a single city (marker). Two cities also fails sometimes. I'd suggest putting an error catch in there to not try to order the small linkage groups.

**Comment by greymonroe** [comment]
Makes sense. Yeah, not a big deal because generally you won’t be looking to order 1 marker chromosomes but thought it was worth noting and provides a reason to get under the hood in the code to fix. An error catch is a good idea.

**Comment by Kopiks** [comment]
Hello! Have you managed to fix this issue?

**Comment by LovellHAGSC** [comment]
Quick fix is to just subset your cross to chromosomes with > 1 marker.
`> new.cross <- subset(cross, chr = nmar(cross) > 1)`

--
John T. Lovell, Ph.D.
Computational Biologist, Genome Sequencing Center
HudsonAlpha Institute for Biotechnology
